mempool/src/pool/tests.rs

```diff
@@ -1879,3 +1879,88 @@ async fn no_empty_bags_in_indices(#[case] seed: Seed) -> anyhow::Result<()> {
     mempool.store.assert_valid();
     Ok(())
 }
+
+struct TestTxAccumulator {
+    txs: Vec<SignedTransaction>,
+    total_size: usize,
+    target_size: usize,
+}
+
+impl TestTxAccumulator {
+    fn new(target_size: usize) -> Self {
+        Self {
+            txs: Vec::new(),
+            total_size: 0,
+            target_size,
+        }
+    }
+}
+
+impl TransactionAccumulator for TestTxAccumulator {
+    fn add_tx(&mut self, tx: SignedTransaction) {
+        self.total_size += tx.encoded_size();
+        self.txs.push(tx);
+    }
+
+    fn done(&self) -> bool {
+        self.target_size <= self.total_size
```

Comment:
This kind of condition and possible bad use for it, kind of... worries me. This can easily cause an overflow over the limit. We have to be careful how to set the condition. If the limit is 1 MB and you do this check, basically you're guaranteed with a very high probability to be over 1 MB. The probability to hit exactly 1 MB is around 1byte/average_tx_size, which is a very low probability. 

Proof to this is in your test below, where you set the limit to 1 byte, but you get 1 transaction larger than that limit.

This has to be carefully engineered to not cause this problem, where the size-limit is a hard limit that'll never be exceeded.

Reply:
Fixed.